botocore/utils.py

```diff
@@ -475,7 +477,6 @@ def _fetch_metadata_token(self):
                     raise InvalidIMDSEndpointError(endpoint=url, error=e)
                 else:
                     raise
-        return None
```

Comment:
Was there a reason we removed this?

Reply:
A misunderstanding of best practice.  Since this is an implicit return, I thought it was more idiomatic to remove it; I see now that since there's another branch that does return something, it's better to have it here.  Thank you